tchannel/errors.py

```diff
@@ -35,97 +42,107 @@ class ProtocolError(TChannelError):
         'tracing',
     )
 
+    code = None
+
     def __init__(
         self,
-        code,
-        description,
+        description=None,
         id=None,
         tracing=None,
     ):
         super(TChannelError, self).__init__(description)
-        self.code = code
         self.tracing = tracing
         self.id = id
         self.description = description
 
+    @classmethod
+    def from_code(cls, code, **kw):
+        return {
+            TIMEOUT: TimeoutError,
+            CANCELED: CanceledError,
+            BUSY: BusyError,
+            DECLINED: DeclinedError,
+            UNEXPECTED_ERROR: UnexpectedError,
+            BAD_REQUEST: BadRequestError,
+            NETWORK_ERROR: NetworkError,
+            UNHEALTHY: UnhealthyError,
+            FATAL: FatalProtocolError,
+        }[code](**kw)
```

Comment:
plz move this dict to `ALL` under the glossary above in case it needs to be reused